azure/functions/decorators/utils.py

```diff
@@ -8,10 +8,8 @@
 from typing import TypeVar, Optional, Union, Iterable, Type, Callable
 
 T = TypeVar("T", bound=Enum)
-SNAKE_CASE_RE = re.compile(r'^([a-z]+\d*_[a-z\d_]*|_+[a-z\d]+[a-z\d_]*)$',
-                           re.IGNORECASE)
-WORD_RE = re.compile(r'^([a-z]+\d*)$', re.IGNORECASE)
-
+SNAKE_CASE_RE = re.compile(r'^([a-zA-Z]+\d*_[a-zA-Z\d_]*|_+[a-zA-Z\d]+[a-zA-Z\d_]*)$')
+WORD_RE = re.compile(r'^([a-zA-Z]+\d*)$')
```

Comment:
I actually think both of these could be simplified/combined into a single shorter regular expression, `(_[a-zA-Z\d]|[a-zA-Z])\w*`, but that would be a larger change and perhaps there's a reason for wanting to separate them.

Reply:
Thanks, @pamelafox.
@YunchuWang - We can take this experience to simplify out regexes.